### ISSUE TITLE
Screen Pipe does not run without sudo (NixOS)

### ISSUE THREAD

**Issue by YoussefZayed** [mediar-ai/screenpipe#406]
I installed screen pipe in nix os following by adding it to my configuration packages but it does not run without sudo and gives this log.

```
[2024-10-02T14:36:33Z WARN  screenpipe] Screenpipe hasn't been extensively tested on this OS. We'd love your feedback!
Would love your feedback on the UX, let's a 15 min call soon:
https://cal.com/louis030195/screenpipe
ALSA lib pcm_dmix.c:1000:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_dmix.c:1000:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_route.c:878:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:878:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:878:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:878:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:878:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:878:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_dmix.c:1000:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_dmix.c:1000:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_dmix.c:1000:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_dmix.c:1000:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_dmix.c:1000:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_route.c:878:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:878:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:878:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:878:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:878:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:878:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_dmix.c:1000:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_dmix.c:1000:(snd_pcm_dmix_open) unable to open slave
[2024-10-02T14:36:33Z INFO  screenpipe] Available audio devices:
[2024-10-02T14:36:33Z INFO  screenpipe]   sysdefault (input)
[2024-10-02T14:36:33Z INFO  screenpipe]   pipewire (input)
[2024-10-02T14:36:33Z INFO  screenpipe]   default (input)
[2024-10-02T14:36:33Z INFO  screenpipe]   sysdefault:CARD=C1 (input)
[2024-10-02T14:36:33Z INFO  screenpipe]   sysdefault:CARD=Generic_1 (input)
[2024-10-02T14:36:33Z INFO  screenpipe]   front:CARD=Generic_1,DEV=0 (input)
[2024-10-02T14:36:33Z INFO  screenpipe]   surround40:CARD=Generic_1,DEV=0 (input)
[2024-10-02T14:36:33Z INFO  screenpipe]   surround51:CARD=Generic_1,DEV=0 (input)
[2024-10-02T14:36:33Z INFO  screenpipe]   surround71:CARD=Generic_1,DEV=0 (input)
[2024-10-02T14:36:33Z INFO  screenpipe]   sysdefault:CARD=Snowball (input)
[2024-10-02T14:36:33Z INFO  screenpipe]   front:CARD=Snowball,DEV=0 (input)
[2024-10-02T14:36:33Z INFO  screenpipe]   pipewire (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   default (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   hdmi:CARD=NVidia,DEV=0 (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   hdmi:CARD=NVidia,DEV=1 (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   hdmi:CARD=NVidia,DEV=2 (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   hdmi:CARD=NVidia,DEV=3 (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   hdmi:CARD=Generic,DEV=0 (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   hdmi:CARD=Generic,DEV=1 (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   hdmi:CARD=Generic,DEV=2 (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   hdmi:CARD=Generic,DEV=3 (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   sysdefault:CARD=Generic_1 (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   front:CARD=Generic_1,DEV=0 (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   surround40:CARD=Generic_1,DEV=0 (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   surround51:CARD=Generic_1,DEV=0 (output)
[2024-10-02T14:36:33Z INFO  screenpipe]   surround71:CARD=Generic_1,DEV=0 (output)
[2024-10-02T14:36:33Z INFO  screenpipe_audio::core] Using default input device: default
[2024-10-02T14:36:33Z INFO  screenpipe_audio::core] Using default output device: default
[2024-10-02T14:36:33Z INFO  screenpipe] Using audio devices:
[2024-10-02T14:36:33Z INFO  screenpipe]   default (input)
[2024-10-02T14:36:33Z INFO  screenpipe]   default (output)
[2024-10-02T14:36:33Z INFO  screenpipe_server::db] Migrations executed successfully.
[2024-10-02T14:36:33Z INFO  screenpipe] Database initialized, will store files in /home/yohguy/.screenpipe
[2024-10-02T14:36:33Z INFO  screenpipe] Server started on http://localhost:3030



                                            _          
   __________________  ___  ____     ____  (_____  ___ 
  / ___/ ___/ ___/ _ \/ _ \/ __ \   / __ \/ / __ \/ _ \
 (__  / /__/ /  /  __/  __/ / / /  / /_/ / / /_/ /  __/
/____/\___/_/   \___/\___/_/ /_/  / .___/_/ .___/\___/ 
                                 /_/     /_/           



Build AI apps that have the full context
Open source | Runs locally | Developer friendly


You are using local processing. All your data stays on your computer.
[2024-10-02T14:36:33Z INFO  screenpipe_server::core] Recording now
[2024-10-02T14:36:33Z INFO  screenpipe_audio::stt] device = Cpu
[2024-10-02T14:36:33Z INFO  hf_hub] Token file not found "/home/yohguy/.cache/huggingface/token"
[2024-10-02T14:36:33Z INFO  screenpipe_server::server] Starting server on 0.0.0.0:3030
[2024-10-02T14:36:33Z INFO  screenpipe_server::video] Starting new video capture
[2024-10-02T14:36:33Z INFO  screenpipe_server::video] Started capture thread
*** bit out of range 0 - FD_SETSIZE on fd_set ***: terminated
fish: Job 1, 'screenpipe' terminated by signal SIGABRT (Abort)

```

Side note, when I run it with sudo it seems to be working but I don't get a UI. I am using gnome as my desktop manager.

**Comment by louis030195** [comment]
@YoussefZayed are you on wayland?

#63

**Comment by YoussefZayed** [comment]
@louis030195 no I am using X11

**Comment by YoussefZayed** [comment]
@louis030195 Here is my desktop enviroment config file


{ config, pkgs, lib, ... }: {
  services.xserver.enable = true;
  services.xserver.displayManager.gdm.enable = true;
  services.xserver.videoDrivers = [ "nvidia" ];
  services.xserver.desktopManager.gnome.enable = true;

  # Configure keymap in X11
  services.xserver.xkb = {
    layout = "us";
    variant = "";
  };

  systemd.services."getty@tty1".enable = false;
  systemd.services."autovt@tty1".enable = false;
}

**Comment by Clausinho** [comment]
> @louis030195 Here is my desktop enviroment config file
> 
> ```
> { config, pkgs, lib, ... }: {
>   services.xserver.enable = true;
>   services.xserver.displayManager.gdm.enable = true;
>   services.xserver.videoDrivers = [ "nvidia" ];
>   services.xserver.desktopManager.gnome.enable = true;
> 
>   # Configure keymap in X11
>   services.xserver.xkb = {
>     layout = "us";
>     variant = "";
>   };
> 
>   systemd.services."getty@tty1".enable = false;
>   systemd.services."autovt@tty1".enable = false;
> }
> ```

I think you most likely are on wayland.

this is my config snipped and I am on wayland:
```
  services.xserver = {
  # we are actually using wayland
  # see https://github.com/NixOS/nixpkgs/issues/94799
  # Enable the X11 windowing system.
    enable = true;
    # Enable the GNOME Desktop Environment.
    displayManager.gdm.enable = true;
    desktopManager.gnome.enable = true;
    videoDrivers = ["i915" "fbdev" "modesetting"/*"nvidia"*/ ];
    # Configure keymap in X11
    xkb = {
      layout = "de";
      variant = "nodeadkeys";
    };
  };
```

Confirm with the output of `echo $XDG_SESSION_TYPE` which returns wayland for me
see https://ostechnix.com/check-wayland-or-x11-in-linux/

So I am pretty sure this can be closed in favor of #63

**Comment by YoussefZayed** [comment]
@Clausinho Thanks, it does look like I am on wayland, Will close ticket.
